### ISSUE TITLE
'Read and change data on GitHub' permission should not be required

### ISSUE THREAD

**Issue by Daanvm** [refined-bitbucket/refined-bitbucket#119]
Hi, I've always been a happy user of this plugin, thanks for all your hard work!

However, I was a bit disappointed this morning when I saw the following two options:
![screen shot 2018-01-24 at 10 41 33](https://user-images.githubusercontent.com/1397121/35325311-a4ccae48-00f3-11e8-8af3-ab926293e382.png)

It's not only that I don't want to give a plugin access to my GitHub gists, it's also that it is probably not even allowed by my employer to do so. I don't have any experience in developing browser plugins and I don't know if it is possible to make such a permission optional, but that would be awesome!

I'm more than willing to help with this if someone can point me in the right direction.

**Comment by reyronald** [comment]
I can completely relate to that feeling. Every time one of my apps asks me for a new permission, I frown!

First, let me provide some context on why the permission is there, for documentation purposes (I may redirect other users that ask about this here). For the pull request template feature (#106), I needed to let the user provide a URL to a place where the template file was hosted publicly, so that the extension could fetch it there and populate the PR body. There are several sites that provide this service, like http://www.pastebin.com and others, but I ultimately settled on GitHubGist since it's the place where I personally store most of my common use snippets, thought the rest of the user base would be most familiar with, and for another important technical reason that I'll detail below. Whichever site I ended up choosing though, I would have needed to include a new permission in the extension to access contents hosted on that domain anyway.

I know the wording on Chrome is a little intimidating when it says "Read and change your data on ... ", but in reality this permission **does not allow me to access a user's gists** ! Gists are managed through https://gist.github.com/, not https://gist.githubusercontent.com/. The latter is just the public interface that GitHub uses to provide direct links to gists raw contents and nothing more, it requires no authentication and does not set any cookies. There's no way for me to use permissions on that host to list or scrape your personal gists, or make authenticated API calls to any content that you have hosted under your account on any of GitHub services. I can only access gists if I know their URL, and there's no way for me to know that URL unless the user provides it to the extension willingly.

In fact, this makes GithubGists the ideal candidate for this need, since other sites that provide this service actually store the raw contents of their snippets on the same domain where the authentication takes place, which would actually pose the security threat you mentioned. I guess GitHub thought of everything! That's why I love this company. 

Now, that said, what you suggest about the optional permission is totally doable and was actually a miss from me. I should've set it up like that in the first place, I just didn't think of it because of what I explained in the paragraphs above, but I understand some people might still feel uncomfortable about it.

Thank you for bringing it up and I'll publish a patch with the change soon! I'll let you know.
